aspnetcore/security/authentication/add-user-data.md

```diff
@@ -102,32 +127,73 @@ Follow the instruction in [Migrations, UseAuthentication, and layout](xref:secur
 
 Update the `IdentityUser` derived class with custom properties. If you named the project WebApp1, the file is named *Areas/Identity/Data/WebApp1User.cs*. Update the file with the following code:
 
-[!code-csharp[Main](add-user-data/sample-2.2/Areas/Identity/Data/WebApp1User.cs)]
+::: moniker range=">= aspnetcore-3.0"
+
+[!code-csharp[](add-user-data/samples/3.x/SampleApp/Areas/Identity/Data/WebApp1User.cs)]
+
+Properties decorated with the [PersonalData](/dotnet/api/microsoft.aspnetcore.identity.personaldataattribute?view=aspnetcore-3.0) attribute are:
```

Comment:
We remove the specific versions from the API docs links (`?view=aspnetcore-3.0`). Again, they're too hard to maintain release-to-release. The API browser will load the latest version. When the reader knows that they're using an older version, they will (should anyway) select the version in the API docs that they want to see.

You can search the whole topic for `?view=` and then crop off the entire query string on these. In this case, chop the `?view=aspnetcore-3.0` off.)

btw - We use xref cross-links to API docs, but let's not get into those changes on this PR. That's a whole can of worms.